azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1729,6 +1726,36 @@ private void cleanExecutionLogs() {
     }
   }
 
+  /**
+   * Calls executor to dispatch the flow, update db to assign the executor and
+   * in-memory state of executableFlow
+   */
+  private void dispatch(ExecutionReference reference, ExecutableFlow exflow,
+    Executor choosenExecutor) throws ExecutorManagerException {
+    exflow.setUpdateTime(System.currentTimeMillis());
+
+    try {
+      executorLoader.assignExecutor(choosenExecutor.getId(),
```

Comment:
the call to assignExecutor should be placed outside the try catch block, as if that thing fails the filed in DB will not be updated with the executor id, therefore there is no need to reset it.